### ISSUE TITLE
--no-eslintrc fails when `eslint-disable` comment disables an external rule

### ISSUE THREAD

**Issue by mgol** [eslint/eslint#12849]
**Tell us about your environment**

* **ESLint Version:** `6.8.0` or `7.0.0-alpha.0`
* **Node Version:** `12.14.1`
* **npm Version:** `6.13.4`

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
'use strict';

module.exports = {
  root: true,

  parserOptions: {
    ecmaVersion: 2019,
    sourceType: 'script',
  },

  plugins: ['react-hooks'],

  rules: {
    'react-hooks/rules-of-hooks': 'error',
  },
};
```

package.json:
```json
{
  "dependencies": {
    "eslint": "6.8.0",
    "eslint-plugin-react-hooks": "2.3.0"
  }
}
```
(optionally, replace `6.8.0` with `7.0.0-alpha.0` above)

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

This is put in a `test.js` file:
```js
'use strict';

// eslint-disable-next-line react-hooks/rules-of-hooks
var x = 2 + 2;
```

```bash
eslint --no-eslintrc test.js 
```

**What did you expect to happen?**

It should not report any error.

**What actually happened? Please include the actual, raw output from ESLint.**

It prints an error:

```
test.js
  3:1  error  Definition for rule 'react-hooks/rules-of-hooks' was not found  react-hooks/rules-of-hooks

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

I don't know how to fix it yet so I can't promise anything.

My use case is that I want to have a special ESLint run that only verifies whether the built JS files are ES5 so that they work in IE 11. Those files are in `.eslintignore` as I don't want regular ESLint runs to check them - people may have outdated built files, etc. I just want to schedule a separate ESLint run after the build fires, only to be invoked in CI, something like:
```
eslint --no-eslintrc --no-ignore --parser-options='{ "ecmaVersion": 5, "sourceType": "script" }' 'packages/*/dist/**'
```

**Comment by platinumazure** [comment]
Hi @mgol, thanks for the issue.

You should be able to use the `--plugin` CLI option to tell ESLint to load the plugin, since ESLint isn't looking at the config file due to `--no-eslintrc`. Will that work for your use case?

**Comment by mgol** [comment]
@platinumazure Thanks, that works as a workaround. However, I think it's not ideal - when I run ESLint with `--no-eslintrc` I'd rather not have to keep an up to date list of plugins specified in various config files when I don't really intend to use those plugins, I just want `eslint-disable` comments disabling some rules to not cause errors.

**Comment by kaicataldo** [comment]
I personally agree that this feels like a bug. I don't think `eslint-disable-*` comments should cause linting errors.

**Comment by platinumazure** [comment]
@kaicataldo What about the case where someone mistyped the plugin or rule name? It makes sense to show a message in those cases, right?

Of course, they don't necessarily need to be lint messages-- if they could just be warnings, that would be ideal...

**Comment by kaicataldo** [comment]
> What about the case where someone mistyped the plugin or rule name? It makes sense to show a message in those cases, right?

That's true, it does seem useful to display an error to the user in the case.
